modules/reindex/src/main/java/org/opensearch/index/reindex/spi/RemoteReindexExtension.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.reindex.spi;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.index.reindex.BulkByScrollResponse;
+import org.opensearch.index.reindex.ReindexRequest;
+
+public interface RemoteReindexExtension {
```

Comment:
Shall we add some java doc here to clarify the intent of this extension point.

Reply:
Added javadoc